extensions/tn_cache/internal/engine_ops.go

```diff
@@ -391,3 +394,34 @@ func (t *EngineOperations) GetIndexComposed(ctx context.Context, provider, strea
 
 	return indexEvents, nil
 }
+
+// GetCurrentHeight returns the current blockchain height by calling helper_get_current_height
+func (t *EngineOperations) GetCurrentHeight(ctx context.Context) (int64, error) {
+	var height int64
+	var found bool
+
+	engineCtx := t.createEngineContext(ctx, "get_current_height")
```

Comment:
I believe this approach wouldn't work because the `createEngineContext` creates a fake execution context just to query with height being 0, so the values wouldn't be the real height. but the fix is in the other comment about getting from the db directly